### PR TITLE
add nightly build for torchserve-kfs image

### PR DESCRIPTION
**Pull request by jagadeeshi2i**
## Description

Workflow run: https://github.com/jagadeeshi2i/serve/actions/runs/6108738716/job/16578339051
Image Build logs: [torchserve-kfs-nightly.log](https://github.com/pytorch/serve/files/12559212/torchserve-kfs-nightly.log)

Please read our [CONTRIBUTING.md](https://github.com/pytorch/serve/blob/master/CONTRIBUTING.md) prior to creating your first pull request.

Please include a summary of the feature or issue being fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #(issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Feature/Issue validation/testing

Please describe the Unit or Integration tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
Logs for Test A

- [ ] Test B
Logs for Test B

## Checklist:

- [ ] Did you have fun?
- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: kubernetes/kserve/docker_nightly.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    organization = args.organization
+
+    project = "torchserve-kfs-nightly"
+    cpu_version = f"{project}:cpu-{get_nightly_version()}"
+    gpu_version = f"{project}:gpu-{get_nightly_version()}"
+
+    # Build Nightly images and append the date in the name
+    try_and_handle(f"./build_image.sh -n -t {organization}/{cpu_version}", dry_run)
+    try_and_handle(
+        f"./build_image.sh -g -cv cu118 -t {organization}/{gpu_version}",
```
Comment (agunapal) [inline kubernetes/kserve/docker_nightly.py:37]: `-n` option missing for GPU images

**Review by agunapal — CHANGES_REQUESTED** [review]
Can you please share a link to/screenshot of publishing these binaries to your dockerhub?

File: kubernetes/kserve/docker_nightly.py (hunk 1 above)
Comment (jagadeeshi2i) [reply kubernetes/kserve/docker_nightly.py]: fixed

**Comment by jagadeeshi2i** [comment]
> Can you please share a link to/screenshot of publishing these binaries to your dockerhub?

Workflow run: https://github.com/jagadeeshi2i/serve/actions/runs/6108738716/job/16578339051

File: kubernetes/kserve/docker_nightly.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    organization = args.organization
+
+    project = "torchserve-kfs-nightly"
+    cpu_version = f"{project}:cpu-{get_nightly_version()}"
+    gpu_version = f"{project}:gpu-{get_nightly_version()}"
+
+    # Build Nightly images and append the date in the name
+    try_and_handle(f"./build_image.sh -n -t {organization}/{cpu_version}", dry_run)
+    try_and_handle(
+        f"./build_image.sh -g -t {organization}/{gpu_version}",
```
Comment (agunapal) [inline kubernetes/kserve/docker_nightly.py:37]: I still see -n missing for GPU missing? @jagadeeshi2i

**Review by agunapal — CHANGES_REQUESTED** [review]

File: kubernetes/kserve/build_image.sh [hunk 3]
```diff
@@ -23,14 +28,32 @@ do
...
           shift
           ;;
+        -n|--nightly)
+          BUILD_NIGHTLY=true
+          shift
+          ;;
         -t|--tag)
-          DOCKER_TAG="$2"
+          CUSTOM_TAG="$2"
+          USE_CUSTOM_TAG=true
```
Comment (chauhang) [inline kubernetes/kserve/build_image.sh:37]: Please add documentation steps for usage for this. eg what will be the actual command line to generate the GPU kserve image?

**Review by chauhang — DISMISSED** [review]
Please see comments inline and attach test logs

**Comment by jagadeeshi2i** [comment]
> Please see comments inline and attach test logs

Added logs for image building with commads

File: kubernetes/kserve/build_image.sh (hunk 3 above)
Comment (jagadeeshi2i) [reply kubernetes/kserve/build_image.sh:37]: This change does not affect the cli parameters and will not require doc change.

**Review by agunapal — APPROVED** [review]
LGTM

**Review by msaroufim — APPROVED** [review]

**Pull request merged**